facilitator/src/transport/s3.rs

```diff
@@ -192,13 +198,33 @@ impl Transport for S3Transport {
         info!("get {}/{} as {:?}", self.path, key, self.iam_role);
         let mut runtime = basic_runtime()?;
         let client = (self.client_provider)(&self.path.region, self.iam_role.clone())?;
-        let get_output = runtime
-            .block_on(client.get_object(GetObjectRequest {
+
+        let mut retries = 0;
+        let get_output = loop {
+            match runtime.block_on(client.get_object(GetObjectRequest {
                 bucket: self.path.bucket.to_owned(),
                 key: [&self.path.key, key].concat(),
                 ..Default::default()
-            }))
-            .context("error getting S3 object")?;
+            })) {
+                Ok(get_output) => break Ok(get_output),
```

Comment:
Style nit: The `Ok(get_output)` and `Err(err)` branches of this match can consolidated into a single catch-all `result => result` branch that comes after the `Err(RusotoError..)` one.

Reply:
Extremely cool, I did not know you could do that!